hazelcast/src/test/java/com/hazelcast/cache/JCacheConcurrencyTest.java

```diff
@@ -0,0 +1,167 @@
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import javax.cache.CacheManager;
+import javax.cache.configuration.CompleteConfiguration;
+import java.net.URI;
+import java.util.Properties;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReferenceArray;
+
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.assertNull;
+import static junit.framework.Assert.assertTrue;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(QuickTest.class)
+public class JCacheConcurrencyTest
+        extends HazelcastTestSupport {
+
+    private static final int CONCURRENCY_LEVEL = 100;
+
+    private TestHazelcastInstanceFactory factory;
+    private HazelcastInstance hz1;
+    private HazelcastInstance hz2;
+
+    private HazelcastServerCachingProvider cachingProvider1;
+    private HazelcastServerCachingProvider cachingProvider2;
+
+    @Before
+    public void init() {
+        factory = new TestHazelcastInstanceFactory(2);
+        hz1 = factory.newHazelcastInstance();
+        hz2 = factory.newHazelcastInstance();
+        cachingProvider1 = HazelcastServerCachingProvider.createCachingProvider(hz1);
+        cachingProvider2 = HazelcastServerCachingProvider.createCachingProvider(hz2);
+    }
+
+    @After
+    public void tear() {
+        cachingProvider1.close();
+        cachingProvider2.close();
+        factory.shutdownAll();
+    }
+
+    @Test
+    public void test_concurrect_cachemanager_creation()
+            throws Exception {
+        final URI uri = new URI(randomString());
+        final Properties properties = HazelcastCachingProvider.propertiesByInstanceName(hz1.getName());
+
+        final Semaphore startSemaphore = new Semaphore(CONCURRENCY_LEVEL);
+        final CountDownLatch endLatch = new CountDownLatch(CONCURRENCY_LEVEL);
+        final AtomicReferenceArray<CacheManager> managers = new AtomicReferenceArray<CacheManager>(CONCURRENCY_LEVEL);
+
+        startSemaphore.acquire(100);
+        for (int i = 0; i < CONCURRENCY_LEVEL; i++) {
+            final int index = i;
+            new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        // Sync all threads
+                        startSemaphore.acquire();
+
+                        // Create CacheManagers
+                        CacheManager cacheManager = cachingProvider1.getCacheManager(uri, null, properties);
+                        managers.set(index, cacheManager);
+
+                    } catch (Exception e) {
+                        e.printStackTrace();
```

Comment:
Is this desirable? If you have an exception, your test won't fail. If you are expecting a certain exception, you need to catch the specific exception and name the variable as ignore/expected.